pkg/modfilter/modfilter.go

```diff
@@ -0,0 +1,192 @@
+package modfilter
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/gomods/athens/pkg/config/env"
+)
+
+var (
+	pathSeparator = "/"
+)
+
+// ModFilter is a filter of modules
+type ModFilter struct {
+	root ruleNode
+}
+
+// NewModFilter creates new filter based on rules defined in a configuration file
+// Configuration consists of two operations + for include and - for exclude
+// e.g.
+//    - github.com/a
+//    + github.com/a/b
+// will communicate all modules except github.com/a and its children, but github.com/a/b will be communicated
+// example 2:
+//   -
+//   + github.com/a
+// will exclude all items from communication except github.com/a
+func NewModFilter() *ModFilter {
```

Comment:
can you name this `NewFilter()`? then the call will be `mod.NewFilter()` which will be less redundant

Reply:
also, do you mind adding a note here that this struct is not concurrency safe? not sure if that matters right now, but it'll be nice to have that later on at least as a reminder 😄